### PR TITLE
Zammad: remove unique URL & add webhook token

### PR DESCRIPTION
**Pull request by svenseeberg**
### Short description
Remove the unique constraint for Zammad URLs and add an authentication token for Zammad webhooks.

### Side effects
- The production database of Malte already has the constraint removed manually. The zammad_url entries have to be removed temporarily, this migration has to be applied, and the zammad_url fields have to be restored.

### Resolved issues

Fixes: #3192

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Comment by svenseeberg** [comment]
We should probably create a release after this has been merged.

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -2334,10 +2334,10 @@ msgstr ""
...
 #: cms/forms/regions/region_form.py
 msgid ""
-"A Zammad URL and Access Token are required in order to enable the Integreat "
-"Chat."
+"A Zammad URL, Zammad Webhook Token and Access Token are required in order to "
+"enable the Integreat Chat."
 msgstr ""
-"Eine Zammad URL und ein Zugangstoken sind erforderlich um, den Integreat "
+"Eine Zammad URL, ein Zammad Webhook Token und ein Zugangstoken sind erforderlich um, den Integreat "
 "Chat aktivieren."
```
Comment (MizukiTemma) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:2341]: I'm not native speaker but the German translation seems to have some typo 👀 

```suggestion
"Eine Zammad URL, ein Zammad Webhook Token und ein Zugangstoken sind erforderlich, um den Integreat "
"Chat zu aktivieren."
```

File: integreat_cms/cms/models/regions/region.py [hunk 2]
```diff
@@ -418,9 +419,7 @@ class Region(AbstractBaseModel):
     zammad_url = models.URLField(
         max_length=256,
         blank=True,
-        null=True,
```
Comment (MizukiTemma) [inline integreat_cms/cms/models/regions/region.py:421]: `IntegrityError` occurs when a region is created without Zammad-URL 👀

**Review by MizukiTemma — CHANGES_REQUESTED** [review]
Thank you for the PR 👍 I have two comments.

File: integreat_cms/cms/models/regions/region.py (hunk 2 above)
Comment (svenseeberg) [reply integreat_cms/cms/models/regions/region.py:421]: wild ... thanks :)

Comment (svenseeberg) [reply integreat_cms/cms/models/regions/region.py:421]: Yeah okay, we had our self defined clean method that returned `None`.

**Comment by svenseeberg** [comment]
It seems the migration does not fully work on its own as it is one large transaction. On integreat-test.tuerantuer.org I had to apply the steps manually:
```
root@integreat-test ~ # integreat-cms-cli sqlmigrate cms 0110
BEGIN;
--
-- Add field zammad_webhook_token to region
--
ALTER TABLE "cms_region" ADD COLUMN "zammad_webhook_token" uuid DEFAULT 'e5961270779e45f481ed7c4e82354199'::uuid NOT NULL;
ALTER TABLE "cms_region" ALTER COLUMN "zammad_webhook_token" DROP DEFAULT;
--
-- Alter field zammad_url on region
--
ALTER TABLE "cms_region" DROP CONSTRAINT "cms_region_zammad_url_a14c5e8a_uniq";
DROP INDEX IF EXISTS "cms_region_zammad_url_a14c5e8a_like";
--
-- Raw Python operation
--
-- THIS OPERATION CANNOT BE WRITTEN AS SQL
--
-- Alter field zammad_url on region
--
ALTER TABLE "cms_region" ALTER COLUMN "zammad_url" SET DEFAULT '';
UPDATE "cms_region" SET "zammad_url" = '' WHERE "zammad_url" IS NULL; SET CONSTRAINTS ALL IMMEDIATE;
ALTER TABLE "cms_region" ALTER COLUMN "zammad_url" SET NOT NULL;
ALTER TABLE "cms_region" ALTER COLUMN "zammad_url" DROP DEFAULT;
COMMIT;
```

```
1 root@integreat-test ~ # su postgres                                                                :(
postgres@integreat-test:/root$ cd
postgres@integreat-test:~$ psql integreat
psql (15.8 (Debian 15.8-0+deb12u1))
Type "help" for help.

integreat=# ALTER TABLE "cms_region" ADD COLUMN "zammad_webhook_token" uuid DEFAULT 'e5961270779e45f481ed7c4e82354199'::uuid NOT NULL;
ALTER TABLE
integreat=# ALTER TABLE "cms_region" ALTER COLUMN "zammad_webhook_token" DROP DEFAULT;
ALTER TABLE
integreat=# ALTER TABLE "cms_region" DROP CONSTRAINT "cms_region_zammad_url_a14c5e8a_uniq";
ALTER TABLE
integreat=# DROP INDEX IF EXISTS "cms_region_zammad_url_a14c5e8a_like";
DROP INDEX
integreat=# ALTER TABLE "cms_region" ALTER COLUMN "zammad_url" SET DEFAULT '';
ALTER TABLE
integreat=# UPDATE "cms_region" SET "zammad_url" = '' WHERE "zammad_url" IS NULL; SET CONSTRAINTS ALL IMMEDIATE;
UPDATE 176
WARNING:  SET CONSTRAINTS can only be used in transaction blocks
SET CONSTRAINTS
integreat=# ALTER TABLE "cms_region" ALTER COLUMN "zammad_url" SET NOT NULL;
ALTER TABLE
integreat=# ALTER TABLE "cms_region" ALTER COLUMN "zammad_url" DROP DEFAULT;
ALTER TABLE
integreat=# SELECT zammad_url FROM cms_region;
integreat=# 
```

**Pull request merged**